src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1012,6 +1015,8 @@ protected void MarkAssembly (AssemblyDefinition assembly)
 
 			foreach (ModuleDefinition module in assembly.Modules)
 				LazyMarkCustomAttributes (module, module);
+
+			_context.Suppressions = new UnconditionalSuppressMessageAttributeState (assembly);
```

Comment:
This is probably not right. The `_context` is global - for all assemblies, and here we're assigning a state which is for one assembly only. Also - what if there already is another state in that variable? MarkStep doesn't work one assembly at a time - it jumps randomly between assemblies.

We either need to store the suppressions per-assembly or have some kind of global storage keyed of the metadata token providers.